### DISCUSSION TITLE
Show global keys in mini.clue

### DISCUSSION THREAD

**Discussion by azdanov** [nvim-mini/mini.nvim#674] [Q&A] [answered]
How can I show global keys in mini.clue?

When using which-key.nvim I can go back (`<BS>`) all the way to the root:
![116439871-df9ee480-a804-11eb-9529-800e167db65c](https://github.com/echasnovski/mini.nvim/assets/6123841/668ac3e0-67bf-4d42-9831-e3f6ee70e4ac)

**Comment by echasnovski** [comment]
What do you mean by "global keys"? 'mini.clue' starts its key query process [strictly on opt-in triggers](https://github.com/echasnovski/mini.nvim/blob/1cd8d7e8318071eddc536815f5cafe50b012ac29/doc/mini-clue.txt#L10) while [`<BS>` can also be used to remove previous entry](https://github.com/echasnovski/mini.nvim/blob/1cd8d7e8318071eddc536815f5cafe50b012ac29/doc/mini-clue.txt#L191-L192). Pressing `<BS>` can remove all the until the initial trigger (if it is several keys, they are treated as single entry).

**Reply by azdanov** [reply]
I'm not sure how to call them, but by "global keys" I meant built-in keys. For example, if I only press `d` or `a` to show what other follow-up keys are available. Or to show if I didn't have any keys pressed, what keys are available, because I have a key that is mapped to `J` letter, but I'm not sure how to display the clue for it.

**Reply by echasnovski** [reply, marked as answer]
I see.

This is not supported right away from the box by 'mini.clue'. As I said earlier, it works *only* for and starting with **opt-in** triggers. For example, `<Leader>`, `g`, etc. Here is an [example](https://github.com/echasnovski/mini.nvim/blob/1cd8d7e8318071eddc536815f5cafe50b012ac29/doc/mini-clue.txt#L243).

It also only shows info about mappings automatically or [manually set up clues](https://github.com/echasnovski/mini.nvim/blob/1cd8d7e8318071eddc536815f5cafe50b012ac29/doc/mini-clue.txt#L325). This matters for your examples:
- `d` usually starts an Operator-pending mode and usually doesn't have mappings starting with it (so clues probably won't get shown);
- `a` inside Visual or Operator-pending mode may not have many mappings defined after it. For example, 'mini.ai' treats `a` as key itself and doesn't create separate mappings for every textobject it implements (for good reasons of allowing easier buffer-local config and not relying on 'timeoutlen' option).

This all might have sounded a bit technical, but the bottom line is that 'mini.clue' is not specifically designed for the task you're wondering. To a certain extent it can be achieved, but I'd suggest to not embark on this journey.

**Reply by azdanov** [reply]
Thank you for your time and a clear answer 🙂
